### PR TITLE
feat(objectionary#3038): add trufflehog-oss workflow

### LINKED ISSUE #3038: Let's add trufflehog-oss GHA to the organization
This one: https://github.com/marketplace/actions/trufflehog-oss

Just to be safe.

### PR DESCRIPTION
**Pull request by slry**
@yegor256 take a look please

Closes #3038

---

## PR-Codex overview
This PR updates the license information and adds a GitHub Actions workflow for trufflehog security scanning.

### Detailed summary
- Updated license to MIT with copyright notice
- Added GitHub Actions workflow for trufflehog security scanning
- Workflow triggers on push and pull requests to the master branch
- Workflow runs trufflehog security scanning with extra_args set to --only-verified

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: .github/workflows/trufflehog-oss.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+'on':
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
Comment (yegor256) [inline .github/workflows/trufflehog-oss.yml:34]: @slry it's better to use `ubuntu-22.04`

File: .github/workflows/trufflehog-oss.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+name: trufflehog-oss
+'on':
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  test:
```
Comment (yegor256) [inline .github/workflows/trufflehog-oss.yml:33]: @slry better use `trufflehog` name here

**Comment by slry** [comment]
@yegor256 can we merge this?

**Comment by slry** [comment]
@yegor256 can we merge this?

**Comment by yegor256** [comment]
@slry thanks!

**Pull request merged**